lib/xenops_server.ml

```diff
@@ -2338,6 +2339,18 @@ module VM = struct
          let vm_id = VGPU_DB.vm_of vgpu_id in
          match context.transferred_fd with
          | Some transferred_fd ->
+
+           (* prevent vgpu-migration from pre-Jura to Jura and later *)
+           if not (List.mem_assoc cookie_vgpu_migration cookies) then
+               begin
+                  (* only Jura and later hosts send this cookie; fail the migration from pre-Jura hosts *)
+                  let msg = Printf.sprintf "VM.migrate: version of sending host incompatible with receiving host: no cookie %s" cookie_vgpu_migration in
+                  Xenops_migrate.(Handshake.send ~verbose:true transferred_fd (Handshake.Error msg));
+                  debug "VM.receive_vgpu: Synchronisation point 1-vgpu ERR %s" msg;
+                  raise (Internal_error msg)
```

Comment:
Shoud this be a proper `Xapi_error.Server_error`? Internal errors are for xapi messups, here I think we need to properly inform the user and make XenCenter show an appropriate message

Reply:
This is in Xenopsd, not Xapi - so this would have to be propagated.

Reply:
Yes, it needs to be added to `xcp-idl` and then go to `xapi`. I am happy if we decide to merge it but we create a new ticket to deal with the error properly

Reply:
I agree - we merge and create tickets to handle the error better.

Reply:
created CP-26159 to propagate a more friendly error message